File: easy/1137. N-th Tribonacci Number/1137. N-th Tribonacci Number.py
from functools import cache


class Solution:
    def tribonacci(self, n: int) -> int:
        @cache
        def run(n: int) -> int:
            if n == 0:
                return 0
            if n <= 2:
                return 1
            return run(n - 1) + run(n - 2) + run(n - 3)

        return run(n)

File: easy/1137. N-th Tribonacci Number/1137. N-th Tribonacci Number.ts
function tribonacci(n: number): number {
    if (!n)
        return 0

    const dp = new Array<number>(n + 1).fill(0)
    dp[1] = dp[2] = 1

    for (let i = 3; i < dp.length; i++) {
        dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3]
    }

    return dp[dp.length - 1]
};

console.log(tribonacci(4))
console.log(tribonacci(25))